adyen.php

```diff
@@ -975,4 +910,109 @@ private function createCustomerMessage($message, CustomerThread $customerThread)
+                        ),
+                        'threeDSProcessUrl' => $this->context->link->getModuleLink(
+                            $this->name, 'ThreeDSProcess',
+                            array(), true
+                        ),
+                        'prestashop16' => true,
+                        'oneClickPaymentMethod' => json_encode($storedCard),
+                        'recurringDetailReference' => $storedCard['recurringDetailReference'],
+                        'name' => $storedCard['name'],
+                        'number' => $storedCard['storedDetails']['card']['number']
+                    )
+                );
+            }
+            $payments .= $this->display(__FILE__, '/views/templates/front/oneclick.tpl');
+        }
+        return $payments;
+    }
+
+    /**
+     * @param array $paymentMethods
+     *
+     * @return string
+     */
+    private function getLocalPaymentMethods(array $paymentMethods)
+    {
+        $payments = '';
+        foreach ($paymentMethods['paymentMethods'] as $paymentMethod) {
+            $issuerList = [];
+            if (empty($paymentMethod['type']) || $paymentMethod['type'] == 'scheme') {
+                continue;
+            }
+            if (isset($paymentMethod['details'])) {
+                foreach ($paymentMethod['details'] as $paymentMethodDetails) {
+                    if (key_exists('key', $paymentMethodDetails) && $paymentMethodDetails['key'] == 'issuer') {
+                        $issuerList = $paymentMethodDetails['items'];
+                        break;
+                    }
+                }
+            }
+            $this->context->smarty->assign(
+                array(
+                    'locale' => $this->helper_data->getLocale($this->context->language),
+                    'originKey' => $this->helper_data->getOriginKeyForOrigin(),
+                    'environment' => Configuration::get('ADYEN_MODE'),
+                    'issuerList' => json_encode($issuerList),
+                    'paymentMethodType' => $paymentMethod['type'],
+                    'paymentMethodName' => $paymentMethod['name'],
+                    'paymentProcessUrl' => $this->context->link->getModuleLink(
+                        $this->name,
+                        'Payment',
+                        array(),
+                        true
+                    ),
+                    'renderPayButton' => true,
+                )
+            );
+            $payments .= $this->display(__FILE__, '/views/templates/front/local-payment-method.tpl');
+        }
+        return $payments;
+    }
+
+    /**
+     * @return string
+     */
+    private function getStandardPaymentMethod()
+    {
+        $payments = '';
+        $this->context->smarty->assign(
+            array(
+                'locale' => $this->helper_data->getLocale($this->context->language),
+                // no locale in PrestaShop1.6 only languageCode that is en-en but we need en_EN
```

Comment:
should we define a default ?

Reply:
what do you mean with this?

Reply:
You are specifying that for 1.6 the format is incorrect en-en instaed of en_EN should we then not prefine it for 1.6 to en_EN or should we make a converter that makes en-en to en_EN ?

Reply:
I think we need to do this in the getLocale function in the helper_data, I will create a youtrack ticket for this, so we can discuss